### PR TITLE
Improvement/storybook docs

### PR DESCRIPTION
**Pull request by JeanMarcMilletScality**
**Component**:

**Description**:

**Design**:

**Breaking Changes**:

- [] Breaking Changes

---

Closes: #ISSUE_NUMBER

### CODE REVIEW

File: src/lib/components/tooltip/Tooltip.component.tsx [hunk 1]
```diff
@@ -96,6 +96,7 @@ function Tooltip({
         placement,
         middleware: [offset(5), shift(), flip()],
       }).then(({ x, y }) => {
+        console.log('cdc', x, y);
```
Comment (ChengYanJin) [inline src/lib/components/tooltip/Tooltip.component.tsx]: Oooops. ;)

File: stories/Button/Guideline.mdx [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+#### Danger
+
+Use for critical action, typically delete actions
+
+<Story of={ButtonStories.Danger} />
+
+### Links
+
+_to add_
```
Comment (ChengYanJin) [inline stories/Button/Guideline.mdx]: not sure what do we want to provide here ?

File: stories/Button/button.stories.tsx [hunk 3]
```diff
@@ -23,29 +18,43 @@ export default {
   args: {
     onClick: action('Button clicked'),
   },
+  render: ({ icon, ...args }) => {
+    console.log('canvas', <Canvas />);
```
Comment (ChengYanJin) [inline stories/Button/button.stories.tsx]: I think the log can be removed ;)

**Review by ChengYanJin — COMMENTED** [review]
it's nice to have the guidance next to the component ! 
 Just a few nitpicking to address.

File: stories/Select/selectv2.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+        <Select size={size} {...args}></Select>
+      ))}
+    </SelectWrapper>
+  ),
+  args: {
+    children: defaultOptions,
+  },
+};
+
+export const NotEnoughPlaceAtTheBottom = {
```
Comment (ChengYanJin) [inline stories/Select/selectv2.stories.tsx:92]: We have some warnings in the console.  maybe to address it by the way

> Warning: Each child in a list should have a unique "key" prop.

Comment (JeanMarcMilletScality) [reply stories/Select/selectv2.stories.tsx:92]: done 👍

File: src/lib/components/tooltip/Tooltip.component.tsx (hunk 1 above)
Comment (JeanMarcMilletScality) [reply src/lib/components/tooltip/Tooltip.component.tsx]: 😬

Comment (JeanMarcMilletScality) [reply src/lib/components/tooltip/Tooltip.component.tsx]: I deleted it

File: stories/Button/Guideline.mdx [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+} from '@storybook/blocks';
+import { Button } from '../../src/lib/components/buttonv2/Buttonv2.component';
+
+import * as ButtonStories from './button.stories';
+
+<Meta of={ButtonStories} />
+
+# Button
+
+Buttons are used to trigger an action or event when clicked.
```
Comment (JBWatenbergScality) [inline stories/Button/Guideline.mdx]: I would change _clicked_ by "activated" or "triggered". This because you can trigger the action not only by a click but also by a form submission of via a keyboard action on the button

File: stories/Button/Guideline.mdx [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+
+## Size & style
+
+### Style
+
+- Height: 2rem
+- Border-radius: 3px
+- Minimum width: 5rem
```
Comment (JBWatenbergScality) [inline stories/Button/Guideline.mdx]: nit: Maybe we want to unify the way we prensent those style (Border radius without the hyphens or use the hyphen notation for the width), probably we should pick the without hyphens notation to be aligned with the padding

File: stories/Button/Guideline.mdx [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+- Vertical padding: 0.5rem
+- Horizontal padding of 1rem
+- Space between icon & label: 0.5rem
+
+### Button spacing
+
+- Horizontal spacing: 1rem
+- Vertical spacing: 0.75rem
+- Space between 2 buttons: 1.5rem - when there is no space constraint
+
```
Comment (JBWatenbergScality) [inline stories/Button/Guideline.mdx]: @Cuervino shall we add a styling rule when we have icons in the button ?
Maybe also a behavior rule for the "loading" buttons where we replace the icon by some spinners

File: stories/Button/Guideline.mdx [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+Examples: Cancel, Close, Create, Delete, Edit, Learn more, Review, Save, Study, View scores, etc.
+
+Since buttons are action points for the user, button text should also be actionable. \
+Stick to using verbs (Complete, Start/Finish, Search) or a simple verb + noun combination for buttons (Next Page, Submit Post, Learn More). \
+Whichever labeling method you choose, maintain consistency across all of your buttons.
+
+| Do                                                      | Don't                                     |
+| ------------------------------------------------------- | ----------------------------------------- |
+| Use sentence case                                       | Use icons only                            |
+| Use specific action-oriented text for CTAs              | Use punctuation or exclamation points (!) |
```
Comment (JBWatenbergScality) [inline stories/Button/Guideline.mdx]:
```suggestion
| Use specific action-oriented text for CTAs              | Use punctuation or exclamation marks (!) |
```

File: stories/Button/Guideline.mdx [hunk 9]
```diff
@@ -0,0 +1,169 @@
...
+| Capitalize proper nouns                                 | Write “Click here”                        |
+| Check for consistent use across the UI                  |                                           |
+| Reserve enough space for translation to other languages |                                           |
+
+## Variations
+
+The button can be modify in different way to match its usage
```
Comment (JBWatenbergScality) [inline stories/Button/Guideline.mdx]:
```suggestion
The button can be modified in different way to match its usage
```

File: stories/Select/Guideline.mdx [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+  Unstyled,
+  Source,
+} from '@storybook/blocks';
+import { Select } from '../../src/lib/components/selectv2/Selectv2.component';
+
+import * as SelectStories from './selectv2.stories';
+
+<Meta title="Guideline" of={SelectStories} />
+
+# Select
```
Comment (JBWatenbergScality) [inline stories/Select/Guideline.mdx]: I think we are missing some guidelines on when/how to use icons for the options ?

File: stories/guideline/docs-template.mdx [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+
+<Meta title="Guideline" isTemplate />
+
+# Component name
+
+General description
+
+## Style
+
+## How it is used ?
```
Comment (JBWatenbergScality) [inline stories/guideline/docs-template.mdx]:
```suggestion
## How is it used ?
```

File: stories/Button/Guideline.mdx (hunk 2 above)
Comment (Cuervino) [reply stories/Button/Guideline.mdx]: Some buttons could have not an internal action, but be a link to external UIs:
image

However, I'm still not sure of how they should look like. Maybe it would be better to use a hyperlink type of component:
image

Comment (Cuervino) [reply stories/Button/Guideline.mdx]: So it might be removed for now.

File: stories/Button/Guideline.mdx (hunk 5 above)
Comment (Cuervino) [reply stories/Button/Guideline.mdx]: I would use "triggered", as "activated" is to close to the word "active", which is a status.

File: stories/Button/Guideline.mdx (hunk 6 above)
Comment (Cuervino) [reply stories/Button/Guideline.mdx]: Technical question: the height and the border-radius are part of the component, but what about the minimum width? Is it enforced somewhere in the code, or it's more guideline relative?
Moreover, I would say that the minimum-width is valid for standard-size buttons, not for "inline" buttons.

File: stories/Button/Guideline.mdx (hunk 7 above)
Comment (Cuervino) [reply stories/Button/Guideline.mdx]: @JBWatenbergScality 
1. what do you mean by styling rules?
2. Agree for the "loading" rules, though I don't know them.

File: stories/Button/Guideline.mdx [hunk 12]
```diff
@@ -0,0 +1,169 @@
...
+
+import * as ButtonStories from './button.stories';
+
+<Meta of={ButtonStories} />
+
+# Button
+
+Buttons are used to trigger an action or event when clicked.
+
+## Size & style
```
Comment (Cuervino) [inline stories/Button/Guideline.mdx]: In general, the size and style part would need some pictures to have a better understanding (it's called "redlining"), such as : https://images.ctfassets.net/7jw9uvgmirvi/7eFu7zc4oXotX8DzxKZy3j/b5a60e6c561528e91b60fd34f2a040d6/5_button_redline.png
It could be in another iteration.

File: stories/Button/Guideline.mdx [hunk 13]
```diff
@@ -0,0 +1,169 @@
...
+- Horizontal padding of 1rem
+- Space between icon & label: 0.5rem
+
+### Button spacing
+
+- Horizontal spacing: 1rem
+- Vertical spacing: 0.75rem
+- Space between 2 buttons: 1.5rem - when there is no space constraint
+
+## Usage
```
Comment (Cuervino) [inline stories/Button/Guideline.mdx]: I'd remove the do/don't, as we are kind of mixing them.
Here is the proposal:


Button labels should be as short and clear as possible and should describe the action the button performs.

- Use one or two words if possible,
- Remove most prepositions and articles (a, an, the).
- Examples: Cancel, Close, Create, Delete, Edit, Learn more, Review, Save, Study, View scores, etc.,
- Stick to using verbs (Complete, Start, Finish, Search) or a simple verb + noun combination for buttons (Next page, Submit post, Learn more),
- Maintain labelling method consistency across all of your buttons,
- Don't use punctuation or exclamation marks (!),
- And, most importantly, don't write "Click here".



About the sentence case: we are not really using this type of case for buttons, as most often, we capitalize the noun after the verb (Create Folder, Create Node, Edit Location). So we are closer to "Title Case" for the button labels.

File: stories/Button/Guideline.mdx (hunk 9 above)
Comment (Cuervino) [reply stories/Button/Guideline.mdx]: Don't forget dot "." at the end of sentences. :-)

File: stories/Button/Guideline.mdx [hunk 14]
```diff
@@ -0,0 +1,169 @@
...
+
+### Variant
+
+Buttons have a set of predefined variants to use in different context
+
+<Story of={ButtonStories.DefaultButtons} />
+
+#### Primary
+
+Used for the main action. It appears once within a group of buttons (or even a screen).
```
Comment (Cuervino) [inline stories/Button/Guideline.mdx]: Used for the main action. It should only appear once within a group of buttons (or even a screen).

File: stories/Button/Guideline.mdx [hunk 15]
```diff
@@ -0,0 +1,169 @@
...
+#### Primary
+
+Used for the main action. It appears once within a group of buttons (or even a screen).  
+For example, “Continue” in a form.
+
+<Story of={ButtonStories.Primary} />
+
+#### Secondary
+
+Use for the secondary action within a group of buttons.
```
Comment (Cuervino) [inline stories/Button/Guideline.mdx]: I would say "a" secondary action (it could be a lot of different secondary buttons.
And "Used" instead of "Use".

File: stories/Button/Guideline.mdx [hunk 16]
```diff
@@ -0,0 +1,169 @@
...
+
+<Story of={ButtonStories.Secondary} />
+
+#### Outline
+
+Dismissive action, such as the cancel button.
```
Comment (Cuervino) [inline stories/Button/Guideline.mdx]: Used for low-emphasis action, as an alternative to Primary or Secondary buttons.
It's suitable for dismissive action, such as the "Cancel" button.

File: stories/Button/Guideline.mdx [hunk 17]
```diff
@@ -0,0 +1,169 @@
...
+
+<Story of={ButtonStories.Outline} />
+
+#### Danger
+
+Use for critical action, typically delete actions
```
Comment (Cuervino) [inline stories/Button/Guideline.mdx]: "Used", and with a dot ".".

Comment (Cuervino) [reply stories/Button/Guideline.mdx]: Maybe we could add here: Actions triggered by such button often require additional validation, as the critical action cannot be undone.

File: stories/Button/Guideline.mdx [hunk 18]
```diff
@@ -0,0 +1,169 @@
...
+
+<Story of={ButtonStories.Danger} />
+
+### Links
+
+_to add_
+
+### Size
+
+Only 2 size are available for the button component : default and inline
```
Comment (Cuervino) [inline stories/Button/Guideline.mdx]: - "2 sizes".
- I think we discussed about the use-cases of the different sizes already.

File: stories/Button/Guideline.mdx [hunk 19]
```diff
@@ -0,0 +1,169 @@
...
+
+_Usecases of the different sizes ?_
+
+<Story of={ButtonStories.ButtonSizes} />
+
+### States
+
+#### Disabled button
+
+Disabled state only if an action on the current screen can enable it (action possibility).
```
Comment (Cuervino) [inline stories/Button/Guideline.mdx]: I would rephrase, here is my proposal:

When the action linked to a button is not possible, the button should be visually disabled rather than removed from the interface for consistent user interaction. 
A hovering tooltip must be provided to explain the reason behind the button's disabled condition.

File: stories/Button/Guideline.mdx [hunk 20]
```diff
@@ -0,0 +1,169 @@
...
+<Story of={ButtonStories.ButtonDisabled} />
+
+## Group of Button
+
+- Primary buttons placed on the right of a group.
+- Dismissive actions placed on the left. It allows the user to return to the previous screen or step in the process.
+
+Align button groups with the container's right side by default; however, based on the use case and visual balance it can be aligned differently.
+
+<Story of={ButtonStories.SimpleForm} />
```
Comment (Cuervino) [inline stories/Button/Guideline.mdx]: This story is rather strange, and it doesn't follow all the rules we would like to apply.
- not sure about the input components
- the input fields should be aligned
- the button should respect the min-width described earlier (5rem)

File: stories/Button/Guideline.mdx [hunk 21]
```diff
@@ -0,0 +1,169 @@
...
+Universally understood icons work well (ie. print, close, play/pause, save).\
+Use standard icons when their use matches their meaning, or at least the user's intent.\
+Avoid creating new icons for every action, especially infrequently used ones.\
+Avoid using an icon alone in a button. If no label is provide, then use a tooltip.
+
+### Placement
+
+When use with a label, the icon is placed on the left side.\
+The icon should be (approximately) the same height as the text.  
+Seeing the icon first help users to scan the page more easily, except for few cases such as navigation (right arrow).
```
Comment (Cuervino) [inline stories/Button/Guideline.mdx]: I forgot what we meant by: "except for few cases such as navigation (right arrow)."

File: stories/Button/Guideline.mdx [hunk 22]
```diff
@@ -0,0 +1,169 @@
...
+
+<Story of={ButtonStories.ButtonsWithIcon} />
+
+Avoid using an icon without label, used Universally understood icons such as Save or Delete.
```
Comment (Cuervino) [inline stories/Button/Guideline.mdx]: This sentence could be removed, it's already mentioned before.

File: stories/Button/Guideline.mdx [hunk 23]
```diff
@@ -0,0 +1,169 @@
...
+They are low-emphasis, making them ideal for non-primary actions in UIs.  
+They're also effective in space-limited areas like tables or key-value lists.
+
+| Do                                              | Don't                                                                                                        |
+| ----------------------------------------------- | ------------------------------------------------------------------------------------------------------------ |
+| Ensure tooltips are included for clarity        | Confuse them with non-interactive elements like descriptive icons or status indicators                       |
+| Use them for secondary or less crucial actions. | Opt for them over Outline buttons, which are more visibly recognizable as buttons.                           |
+|                                                 | Create too many different instances of ghost buttons; limit them to common actions (e.g. link to other UIs). |
+
+<Story of={ButtonStories.GhostButtons} />
```
Comment (Cuervino) [inline stories/Button/Guideline.mdx]: The buttons should have tooltips on hover, if we want to be consistent with what we described earlier.

File: stories/Select/Guideline.mdx [hunk 24]
```diff
@@ -0,0 +1,74 @@
...
+
+The Select component is a dropdown menu that allows users to choose from a list of options.
+It comes with various features to enhance user experience and adapt to different scenarios.
```
Comment (Cuervino) [inline stories/Select/Guideline.mdx]: Not sure this sentence is needs, it's maybe a little bit to general.

File: stories/Select/Guideline.mdx (hunk 10 above)
Comment (Cuervino) [reply stories/Select/Guideline.mdx]: From what I remember, we don't use icons for statuses (unlike what is displayed in the story), it's more _descriptive_ icons that help the user differentiates the different options (for instance, different kinds of workflows).
And if such icons are present in front of labels, they should be present for _all_ labels.

File: stories/Select/Guideline.mdx [hunk 25]
```diff
@@ -0,0 +1,74 @@
...
+You can use this [playground](?path=/story/components-inputs-select--playground) to build the component
+
+## Size & Style
+
+## Usage
+
+## Variations
+
+The select component come with multiple props allowing customization and adaptation to various scenarios
```
Comment (Cuervino) [inline stories/Select/Guideline.mdx]: "." is needed.

File: stories/Select/Guideline.mdx [hunk 26]
```diff
@@ -0,0 +1,74 @@
...
+### Size variations
+
+By default the Select component width is 20.5rem .
+You can reduce its size by defining the 'size' props to two third, half or one third of the default size :
+
+<Story of={SelectStories.DifferentSizes} />
+
+### Options number
+
+Depending on the number of options provide to the select, different features will be automatically enable.
```
Comment (Cuervino) [inline stories/Select/Guideline.mdx]: "enabled".

File: stories/Select/Guideline.mdx [hunk 27]
```diff
@@ -0,0 +1,74 @@
...
+
+#### No option
+
+When no option are provide, a clear information will be given to the user
```
Comment (Cuervino) [inline stories/Select/Guideline.mdx]: "provided".

File: stories/Select/Guideline.mdx (hunk 10 above)
Comment (Cuervino) [reply stories/Select/Guideline.mdx]: The "selected" status visual information doesn't come with the _checked_ icon, but with a "selected" color.

File: stories/Select/Guideline.mdx [hunk 28]
```diff
@@ -0,0 +1,74 @@
...
+By default the Select component width is 20.5rem .
+You can reduce its size by defining the 'size' props to two third, half or one third of the default size :
+
+<Story of={SelectStories.DifferentSizes} />
+
+### Options number
+
+Depending on the number of options provide to the select, different features will be automatically enable.
+
+#### No option
```
Comment (Cuervino) [inline stories/Select/Guideline.mdx]: I'm not sure that we have such case. With such case, shouldn't it be disabled, rather than active with no option?

File: stories/Select/Guideline.mdx [hunk 29]
```diff
@@ -0,0 +1,74 @@
...
+a search bar is automatically added, enabling users to quickly find and select options.
+
+<Story of={SelectStories.WithSearchBar} />
+
+### Disable state
+
+The Select component can be easily disabled, preventing user interaction.
+This is useful in scenarios where user input is not allowed or required
+
+<Story of={SelectStories.DisabledSelect} />
```
Comment (Cuervino) [inline stories/Select/Guideline.mdx]: The story doesn't show a disabled state.

File: stories/form.stories.tsx [hunk 30]
```diff
@@ -1,19 +1,19 @@
+import React from 'react';
+import { Banner } from '../src/lib/components/banner/Banner.component';
```
Comment (Cuervino) [inline stories/form.stories.tsx:2]: I think the triangle icon in the banner is deprecated, we use the circle now.

File: stories/form.stories.tsx [hunk 31]
```diff
@@ -1,19 +1,19 @@
...
-import { Button } from '../src/lib/components/buttonv2/Buttonv2.component';
 import { Icon } from '../src/lib/components/icon/Icon.component';
 import { Input } from '../src/lib/components/inputv2/inputv2';
-import { Toggle } from '../src/lib/components/toggle/Toggle.component';
-import { Banner } from '../src/lib/components/banner/Banner.component';
-import { Text } from '../src/lib/components/text/Text.component';
 import { Select } from '../src/lib/components/selectv2/Selectv2.component';
-import { iconOptions } from './controls';
+import { Text } from '../src/lib/components/text/Text.component';
+import { Toggle } from '../src/lib/components/toggle/Toggle.component';
```
Comment (Cuervino) [inline stories/form.stories.tsx:13]: The toggle component cannot be triggered in the GUI.

File: stories/guideline/docs-template.mdx (hunk 11 above)
Comment (Cuervino) [reply stories/guideline/docs-template.mdx]: In English, no space between the last letter and the question mark. Should be:
## How is it used?

File: stories/guideline/docs-template.mdx
Comment (Cuervino) [inline stories/guideline/docs-template.mdx:1]: This template is not visible in the gui, I guess?

File: stories/guideline/docs-template.mdx [hunk 32]
```diff
@@ -0,0 +1,43 @@
...
+
+## Variations
+
+### Size variations
+
+What size are used for this component ? When to use ?
+
+### Style Variations
+
+Does this component as variant ? When to use ?
```
Comment (Cuervino) [inline stories/guideline/docs-template.mdx]: Incorrect English.

File: stories/guideline/docs-template.mdx [hunk 33]
```diff
@@ -0,0 +1,43 @@
...
+## How it is used ?
+
+General guideline about the component
+Do & Don'ts ?
+
+## Variations
+
+### Size variations
+
+What size are used for this component ? When to use ?
```
Comment (Cuervino) [inline stories/guideline/docs-template.mdx]: "sizes", "When to use them?"

File: stories/guideline/docs-template.mdx [hunk 34]
```diff
@@ -0,0 +1,43 @@
...
+
+### Style Variations
+
+Does this component as variant ? When to use ?
+
+### Status Variations
+
+Does this component can be disable ? Why ?
```
Comment (Cuervino) [inline stories/guideline/docs-template.mdx]: Incorrect English.

**Review by Cuervino — DISMISSED** [review]
Nice structure, massive PR, some polishing. :-)

File: stories/Button/Guideline.mdx (hunk 13 above)
Comment (JeanMarcMilletScality) [reply stories/Button/Guideline.mdx]: Should I add the title case in the list ? something like : - Capitalize the noun after the verb (Create Folder, Create Node, Edit Location).

File: stories/Button/Guideline.mdx (hunk 21 above)
Comment (JeanMarcMilletScality) [reply stories/Button/Guideline.mdx]: Something like a next page button?
image

File: stories/Button/Guideline.mdx (hunk 13 above)
Comment (Cuervino) [reply stories/Button/Guideline.mdx]: yes, good idea!

File: stories/Button/Guideline.mdx (hunk 21 above)
Comment (Cuervino) [reply stories/Button/Guideline.mdx]: yes, but I still don't get what is has to do with the rest of the sentence :-)

Comment (JeanMarcMilletScality) [reply stories/Button/Guideline.mdx]: I think it says that in this case the arrow should be after the label and not before, but this is not possible in the current version of the button

File: stories/Button/Guideline.mdx (hunk 6 above)
Comment (JBWatenbergScality) [reply stories/Button/Guideline.mdx]: The min width can be and should be enforced in the code IMO.

File: stories/Button/Guideline.mdx (hunk 7 above)
Comment (JBWatenbergScality) [reply stories/Button/Guideline.mdx]: For styling: I mean spacing between text and icon, baseline alignements, impact on accessibility. Can we have 2 icons on a single button (at left and right of the text, probably this shall be forbidden ), where do we prefer to put the icon (left or right of the text?, for the right case it seems to be only valid for the 'continue' button with the right arrow) 

For behavior:
- loading behavior
- disabled behavior
- if there is an icon next to the text is the icon read outloud by screen readers in addition to the text ?

File: stories/Button/Guideline.mdx (hunk 12 above)
Comment (JBWatenbergScality) [reply stories/Button/Guideline.mdx]: I think there is a way to get those from figma "dev" mode. In addition there is a storybook extensions that allow us to include panels from figma, maybe we could use it to import those directly from there ? It however brings some questions because our figma file is private and core-ui public.

File: stories/guideline/docs-template.mdx
Comment (JeanMarcMilletScality) [reply stories/guideline/docs-template.mdx:1]: yes, it is not visible

File: stories/Button/Guideline.mdx (hunk 12 above)
Comment (Cuervino) [reply stories/Button/Guideline.mdx]: @JBWatenbergScality I want to extract the Design system from the Figma file, to make it a dedicated file. This one file could be public. 
It makes sense because of:
1. having in the future multiple products fed from the same DS, 
2. performance reason: everything in the same file starts to be too resource-consuming.

File: stories/Button/Guideline.mdx (hunk 21 above)
Comment (Cuervino) [reply stories/Button/Guideline.mdx]: Correct, I think we can keep having an icon only on the left side.

File: stories/Button/Guideline.mdx (hunk 7 above)
Comment (Cuervino) [reply stories/Button/Guideline.mdx]: My proposal

Guidelines that could be added:

---

### Icon-Text Spacing
- **Spacing**: 0.5rem between the text and the icon.

### Icon Usage
- **Placement**: Icons are to be positioned only on the left side of the text.
- **Alignment**: Icons should be vertically center-aligned with the text.

### Accessibility
- **Icons in Button Design**: 
  - Easily recognizable icons help with the quick recognition of a button’s function. If the icon is not clear enough, it loses its purpose, so in such cases, avoid using icons.
  
- **Aria Labels**: 
  - Providing descriptive labels for screen readers is really helpful, especially for buttons that use icons only.

### Button States

#### Loading State
- Display a spinner animation within the button.
- Disable any interactions during the loading process.

#### Disabled State
- Clicks on the button should trigger no action, and the cursor should display a "not-allowed" indicator.
- A tooltip explaining the reason for the button’s disabled state should appear on hover.


---

Note:
The loading state behaviour should be described in the story.

**Review by Cuervino — COMMENTED** [review]
I added few comments, that might be added to the text. Otherwise, let's go with this version.

File: stories/InfoMessage/infomessage.guideline.mdx [hunk 35]
```diff
@@ -0,0 +1,32 @@
+import { Meta, Story, Canvas, Primary, Controls } from '@storybook/blocks';
+
+import * as Stories from './infomessage.stories.tsx';
+
+<Meta name="Guideline" of={Stories} />
+
+# InfoMessage
+
+This component provides complementary information to the user to help him understand what to do.
```
Comment (JBWatenbergScality) [inline stories/InfoMessage/infomessage.guideline.mdx]:
```suggestion
This component provides complementary information to the user to help them understand what to do.
```

File: stories/Select/select.guideline.mdx [hunk 36]
```diff
@@ -0,0 +1,73 @@
...
+  Unstyled,
+  Source,
+} from '@storybook/blocks';
+import { Select } from '../../src/lib/components/selectv2/Selectv2.component';
+
+import * as SelectStories from './selectv2.stories';
+
+<Meta title="Guideline" of={SelectStories} />
+
+# Select
```
Comment (JBWatenbergScality) [inline stories/Select/select.guideline.mdx:16]: @Cuervino maybe for later on we need to think about a loading state for select component requiring async process to get the options

**Review by JBWatenbergScality — COMMENTED** [review]
Overall LGTM, just a tiny comment on users pronoun, prefer usage of a generic inclusive pronoun ;) !

**Review by JBWatenbergScality — APPROVED** [review]
LGTM !
